File: Practice/atcoder/ABC/049/src/c.go
package main

import (
	"fmt"
)

func main() {
	var s string
	fmt.Scan(&s)
	// sの末尾から一致する単語を消していく
	ans := false
	for {
		var tmp = s
		l := len(tmp)
		switch {
		case 0 == l:
			ans = true
			break
		case 6 < l:
			if tmp[len(tmp)-7:] == "dreamer" {
				tmp = tmp[:len(tmp)-7]
				s = tmp
				continue
			}
			fallthrough
		case 5 < l:
			if tmp[len(tmp)-6:] == "eraser" {
				tmp = tmp[:len(tmp)-6]
				s = tmp
				continue
			}
			fallthrough
		case 4 < l:
			if tmp[len(tmp)-5:] == "dream" || tmp[len(tmp)-5:] == "erase" {
				tmp = tmp[:len(tmp)-5]
				s = tmp
				continue
			}
			fallthrough
		default:
			break
		}
		if tmp == s {
			break
		}
	}
	if ans {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}

File: Practice/atcoder/ABC/049/src/c.py
# デジャヴュ。ゼロから始めるのではなく消していってゼロにする
# なぜか1WA
# ほとんど同じはずのこのコードだと通った 意味不明

s = input()
while True:
    if len(s) == 0:
        print("YES")
        exit()
    elif len(s) <= 4:
        print("NO")
        exit()
    else:
        # 長さの処理しなくてもエラーにはならんっぽい
        if s[-7:] == "dreamer":
            s = s[:-7]
        elif s[-6:] == "eraser":
            s = s[:-6]
        elif s[-5:] == "dream" or s[-5:] == "erase":
            s = s[:-5]
        else:
            print("NO")
            break
